dask/array/tests/test_fft.py

```diff
@@ -24,87 +25,129 @@ def key(k):
+    assert_eq(irfft(darr2, 12, axis=0), np.fft.irfft(nparr, 12, axis=0))
 
 
 def test_hfft():
-    assert_eq(hfft(darr), npfft.hfft(nparr))
-    assert_eq(hfft(darr2, axis=0), npfft.hfft(nparr, axis=0))
+    assert_eq(hfft(darr), np.fft.hfft(nparr))
+    assert_eq(hfft(darr2, axis=0), np.fft.hfft(nparr, axis=0))
 
 
 def test_hfft_nkwarg():
-    assert_eq(hfft(darr, 5), npfft.hfft(nparr, 5))
-    assert_eq(hfft(darr, 13), npfft.hfft(nparr, 13))
-    assert_eq(hfft(darr2, 5, axis=0), npfft.hfft(nparr, 5, axis=0))
-    assert_eq(hfft(darr2, 13, axis=0), npfft.hfft(nparr, 13, axis=0))
-    assert_eq(hfft(darr2, 12, axis=0), npfft.hfft(nparr, 12, axis=0))
+    assert_eq(hfft(darr, 5), np.fft.hfft(nparr, 5))
+    assert_eq(hfft(darr, 13), np.fft.hfft(nparr, 13))
+    assert_eq(hfft(darr2, 5, axis=0), np.fft.hfft(nparr, 5, axis=0))
+    assert_eq(hfft(darr2, 13, axis=0), np.fft.hfft(nparr, 13, axis=0))
+    assert_eq(hfft(darr2, 12, axis=0), np.fft.hfft(nparr, 12, axis=0))
 
 
 def test_ihfft():
-    assert_eq(ihfft(darr), npfft.ihfft(nparr))
-    assert_eq(ihfft(darr2, axis=0), npfft.ihfft(nparr, axis=0))
+    assert_eq(ihfft(darr), np.fft.ihfft(nparr))
+    assert_eq(ihfft(darr2, axis=0), np.fft.ihfft(nparr, axis=0))
 
 
 def test_ihfft_n_kwarg():
-    assert_eq(ihfft(darr, 5), npfft.ihfft(nparr, 5))
-    assert_eq(ihfft(darr, 13), npfft.ihfft(nparr, 13))
-    assert_eq(ihfft(darr2, 5, axis=0), npfft.ihfft(nparr, 5, axis=0))
-    assert_eq(ihfft(darr2, 13, axis=0), npfft.ihfft(nparr, 13, axis=0))
-    assert_eq(ihfft(darr2, 12, axis=0), npfft.ihfft(nparr, 12, axis=0))
+    assert_eq(ihfft(darr, 5), np.fft.ihfft(nparr, 5))
+    assert_eq(ihfft(darr, 13), np.fft.ihfft(nparr, 13))
+    assert_eq(ihfft(darr2, 5, axis=0), np.fft.ihfft(nparr, 5, axis=0))
+    assert_eq(ihfft(darr2, 13, axis=0), np.fft.ihfft(nparr, 13, axis=0))
+    assert_eq(ihfft(darr2, 12, axis=0), np.fft.ihfft(nparr, 12, axis=0))
 
 
 def test_fft_consistent_names():
     assert same_keys(fft(darr, 5), fft(darr, 5))
     assert same_keys(fft(darr2, 5, axis=0), fft(darr2, 5, axis=0))
     assert not same_keys(fft(darr, 5), fft(darr, 13))
+
+
+def test_wrap_bad_kind():
+    with pytest.raises(ValueError):
+        fft_wrap(np.ones)
+
+
+@pytest.mark.parametrize("modname", ["numpy.fft", "scipy.fftpack"])
+@pytest.mark.parametrize(
+    "funcname",
+    ["fft", "ifft", "rfft", "irfft", "hfft", "ihfft"]
+)
+@pytest.mark.parametrize("dtype", ["float32", "float64"])
+def test_wrap_ffts(modname, funcname, dtype):
+    fft_mod = pytest.importorskip(modname)
+    try:
+        func = getattr(fft_mod, funcname)
+    except AttributeError:
+        pytest.skip("`%s` missing function `%s`." % (modname, funcname))
+
+    darrc = darr.astype(dtype)
+    darr2c = darr2.astype(dtype)
+    nparrc = nparr.astype(dtype)
+
+    if modname == "scipy.fftpack" and "rfft" in funcname:
+        with pytest.raises(ValueError):
+            wfunc = fft_wrap(func)
```

Comment:
Forgot that this `wfunc` wasn't doing anything here. So added PR ( https://github.com/dask/dask/pull/2114 ) to drop the variable assignment since it is unneeded.